src/transport/net_ib.cc

```diff
@@ -562,8 +563,7 @@ ncclResult_t ncclIbInit(ncclDebugLogger_t logFunction) {
           }
         }
       }
-
-      if (nIbDevs && (ncclSuccess != wrap_ibv_free_device_list(devices))) { ret = ncclInternalError; goto fail; };
+      if (ncclSuccess != wrap_ibv_free_device_list(devices)) { ret = ncclInternalError; goto fail;}
```

Comment:
The original behaviour would avoid returning an error if there were no devices. To maintain that behaviour, you could do this.
```suggestion
      if ((ncclSuccess != wrap_ibv_free_device_list(devices)) && nIbDevs) { ret = ncclInternalError; goto fail; }
```

Reply:
Doing so would also skip your `WARN` below, so you would have to decide whether that's necessary in that case.

Reply:
Ideally we would like to detect if wrap_ibv_free_device_list() is failing/not behaving as expected for some reason ?

Reply:
I'm honestly not sure if it matters for an empty list, but in general I agree we would like to know if this call is failing, as I think it should never fail.